web-client/public/js/graph.js

```diff
@@ -93,74 +93,98 @@ var drawGraph = function (nodes, links, positiveWeights, negativeWeights, sheetT
         return Math.round(totalScale(Math.abs(edge.value)));
     };
 
-    var force = d3.layout.force()
-      .size([width, height])
-      .on("tick", tick)
-      .linkDistance($("#linkDistInput").val())
-      .charge($("#chargeInput").val())
-      .chargeDistance($("#chargeDistInput").val())
-      .gravity($("#gravityInput").val());
+    var simulation = d3.forceSimulation()
+      .force("link", d3.forceLink())
+      .force("charge", d3.forceManyBody())
+      .force("center", d3.forceCenter(width / 2, height / 2));
+
+    var drag = d3.drag()
+      .on("start", dragstart)
+      .on("drag", dragged)
+      .on("end", dragended);
+
+    var dragended = function () {
+        d3.event.stopPropagation();
+    };
+
+    var zoomDragPrevX = 0;
+    var zoomDragPrevY = 0;
+    var zoomDragStarted = function () {
+        zoomDragPrevX = d3.event.x;
+        zoomDragPrevY = d3.event.y;
+        $container.removeClass(CURSOR_CLASSES).addClass("cursorGrabbing");
+        if (!adaptive) {
+            $container.removeClass(CURSOR_CLASSES);
+        }
+    };
+
+    var zoomDragged = function () {
+        if (adaptive) {
+            var scale = 1;
+            if (zoomContainer.attr("transform")) {
+                var string = zoomContainer.attr("transform");
+                scale = 1 / (+(string.substring(string.indexOf("scale(") + 6, string.lastIndexOf(")"))));
+            }
+            zoom.translateBy(zoomContainer, scale * (d3.event.x - zoomDragPrevX), scale * (d3.event.y - zoomDragPrevY));
+            zoomDragPrevX = d3.event.x;
+            zoomDragPrevY = d3.event.y;
+        }
+    };
+
+    var zoomDragEnded = function () {
+        $container.removeClass(CURSOR_CLASSES).addClass("cursorGrab");
+        if (!adaptive) {
+            $container.removeClass(CURSOR_CLASSES);
+        }
+    };
 
-    var drag = force.drag()
-      .origin(function (d) {
-          return d;
-      })
-      .on("dragstart", dragstart);
+    var zoomDrag = d3.drag()
+        .on("start", zoomDragStarted)
+        .on("drag", zoomDragged)
+        .on("end", zoomDragEnded);
```

Comment:
Back to indentation—_this_ looks right to me. If our linter doesn't catch it, then just look out for these types of constructs manually.

Reply:
Yep, linter is good with this. Will take a look at the file again to see if there were any other cases of this that you didn't catch and fix it.